### PR TITLE
fix(graph): store meta rootPath relative to graph file for portability

### PR DESCRIPTION
**Pull request by amondnet**
## Problem

CI encodes repositories with an absolute `rootPath` like `/home/runner/work/soop/soop` and commits the resulting `.soop/graph.meta.json`. When a contributor runs `soop sync` on their machine, the absolute path from CI is meaningless — every lookup that depends on `rootPath` silently resolves to the wrong location.

## Fix

`serializeMeta` and `deserializeMeta` in `packages/graph/src/meta.ts` each gain an optional `graphPath` parameter. When provided:

- **Serialize**: `rootPath` is stored as a path relative to the graph file's directory (e.g., `..` instead of `/home/runner/work/soop/soop`).
- **Deserialize**: the relative `rootPath` is resolved back to an absolute path using `graphPath`'s directory as the base.

Callers that know the on-disk graph path — the encoder's save/load path (`packages/encoder/src/encoder.ts`) and the MCP server/tools (`packages/mcp/src/server.ts`, `packages/mcp/src/tools.ts`, `packages/mcp/src/interactive/encoder.ts`) — now pass it through.

## Backward Compatibility

Meta files that already contain an absolute `rootPath` (e.g., files committed before this change) continue to parse correctly — `deserializeMeta` detects an absolute path and uses it as-is without modification. Callers that omit `graphPath` preserve the previous absolute-path behavior.

## Test Coverage

9 new tests added in `packages/graph/tests/meta.test.ts`:

- Relative encoding when `graphPath` is provided
- Deserialize resolves relative `rootPath` back to absolute
- Legacy meta with absolute `rootPath` parses unchanged
- Cross-machine round-trip (serialize on CI path → deserialize on contributor path)
- Edge cases (no `graphPath`, path with trailing slash, nested graph file)

All 79 existing graph package tests continue to pass.

## Next Steps for Users

The existing `.soop/graph.meta.json` on your branch still contains the old absolute path. It will be automatically regenerated with a portable relative path on the next `soop encode` run (or when CI re-encodes on merge to main). No manual migration is needed.

---
## Summary by cubic
Store `rootPath` in `.soop/graph.meta.json` relative to the graph file so CI-generated meta files work on contributor machines. Also normalize separators for cross-OS portability and handle the graph-in-root “.” case.

- **Bug Fixes**
  - `serializeMeta` and `deserializeMeta` accept an optional `graphPath`; when provided, `rootPath` is saved relative and resolved on read.
  - Normalize path separators to '/' and encode empty relative paths as '.' to support Windows/POSIX and avoid missing roots.
  - `packages/encoder` and `packages/mcp` (server, tools, interactive encoder) now pass the on-disk graph path for JSON and JSONL.
  - Backward compatible: existing meta with absolute `rootPath` parses unchanged; callers without `graphPath` keep prior behavior.
  - Tests in `packages/graph` cover relative encode/decode, legacy meta, cross-machine round-trips, cross-OS separators, same-dir '.', and fixtures avoid `/tmp`.

- **Migration**
  - No action needed. The meta file will regenerate with a relative path on the next `soop encode` or CI run.

<sup>Written for commit 02ec6fe59d521ec0a22ef0badd98e4e86ca3b9a8. Summary will update on new commits.</sup>

### CODE REVIEW

File: packages/graph/src/meta.ts [hunk 1]
```diff
@@ -24,14 +24,39 @@ export function metaPathFor(graphPath: string): string {
...
+  return meta
+}
+
+function encodeRootPath(rootPath: string | undefined, graphPath: string | undefined): string | undefined {
+  if (!rootPath)
+    return undefined
+  const absoluteRoot = path.resolve(rootPath)
+  if (!graphPath)
+    return absoluteRoot
+  return path.relative(path.dirname(path.resolve(graphPath)), absoluteRoot)
```
[the remark this answers is not in the document]
Comment (amondnet) [reply packages/graph/src/meta.ts]: ✅ Applied in [02ec6fe](https://github.com/pleaseai/soop/commit/02ec6fe)

- File: `packages/graph/src/meta.ts`
- Change: `path.relative(...).split(path.sep).join('/') || '.'` — falls back to `.` for the same-directory case, plus normalizes separators for cross-platform portability (combined with @gemini-code-assist's adjacent suggestion).
- Added regression tests: `encodes rootPath as "." when it equals the graph file directory` and `round-trips the same-directory case correctly`.

Good catch on the empty-string round-trip bug. Thanks @cubic-dev-ai!

[the remark this answers is not in the document]
Comment (amondnet) [reply packages/graph/src/meta.ts]: 💡 Implemented with an alternative form in [02ec6fe](https://github.com/pleaseai/soop/commit/02ec6fe)

- Suggested: `.replaceAll('\\\\', '/')`
- Applied: `.split(path.sep).join('/')`
- Reason: `'\\\\'` in JS is a 2-character string (two backslashes), so it would only match UNC-style `\\\\` segments — Windows path separators are single backslashes. Splitting on `path.sep` is a no-op on POSIX and correctly converts single backslashes on Windows. Equivalent outcome for the intent (portable forward-slash storage), and combined with @cubic-dev-ai's `|| '.'` fallback on the same line.

Thanks for flagging the cross-platform concern @gemini-code-assist!

File: packages/graph/src/meta.ts [hunk 2]
```diff
@@ -24,14 +24,39 @@ export function metaPathFor(graphPath: string): string {
...
+/**
+ * Parse a meta object, optionally resolving a relative `rootPath` against
+ * the graph file's directory. Absolute `rootPath` values are preserved as-is
+ * for backward compatibility with legacy meta files.
+ */
+export function deserializeMeta(data: unknown, graphPath?: string): RPGMeta {
+  const meta = RPGMetaSchema.parse(data)
+  if (graphPath && meta.rootPath && !path.isAbsolute(meta.rootPath)) {
+    return { ...meta, rootPath: path.resolve(path.dirname(graphPath), meta.rootPath) }
+  }
```
[the remark this answers is not in the document]
Comment (amondnet) [reply packages/graph/src/meta.ts:51]: ⏭️ Not applied

- Reason: `fs.realpathSync` throws `ENOENT` when the target doesn't exist on disk. `encodeRootPath` / `deserializeMeta` are pure string transforms operating on paths that may not exist yet (newly initialized repos, tests using fixture strings like `/var/repo` or `/Users/alice/projects/soop`, CI runners reading meta before checkout completes). Resolving symlinks here would crash those flows.
- The path-traversal concern doesn't apply at this layer: `meta.ts` doesn't open files, it just rewrites a path string for storage. Symlink resolution (if needed for security) belongs at the file-access boundary in `packages/tools/src/fetch.ts`, not in metadata serialization.
- The separator normalization angle is addressed in the adjacent fix via `.split(path.sep).join('/')` (see reply on line 61).
- The 'reference implementation alignment' guidance is too generic to action without a specific field to remove — happy to revisit if you can point to a concrete divergence.

Thanks for the careful review @gemini-code-assist!
